src/main/java/de/bonndan/nivio/assessment/kpi/KubernetesKPI.java

```diff
@@ -0,0 +1,107 @@
+package de.bonndan.nivio.assessment.kpi;
+
+import de.bonndan.nivio.assessment.Status;
+import de.bonndan.nivio.assessment.StatusValue;
+import de.bonndan.nivio.model.Component;
+import de.bonndan.nivio.model.Label;
+import de.bonndan.nivio.model.Labeled;
+import org.apache.commons.collections.map.SingletonMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.NonNull;
+import org.springframework.util.ObjectUtils;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class KubernetesKPI implements KPI {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KubernetesKPI.class);
+    public static final String IDENTIFIER = "k8s";
+
+    private boolean enabled = true;
+
+    @Override
+    @NonNull
+    public List<StatusValue> getStatusValues(Component component) {
+        if (!(component instanceof Labeled))
+            return new ArrayList<>();
+
+
+        var statusList = new ArrayList<StatusValue>();
+        var counter = new AtomicInteger(0);
+        ((Labeled) component).getLabels(Label.k8s).forEach((key, value) -> {
+            if (!ObjectUtils.isEmpty(value)) {
```

Comment:
You might inverse the if condition to reduce nesting level. Also, the if-branches might only define the "`new StatusValue(...)`" add `statusList.add(...)` could be called at the end of each iteration.